File: cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/ConsentLibException.java
package com.sourcepoint.gdpr_cmplibrary;

public class ConsentLibException extends Exception {

    public String consentLibErrorMessage;

    ConsentLibException(Throwable e, String  message){ super(e); consentLibErrorMessage = message;}
    ConsentLibException(String  message){ super(); consentLibErrorMessage = message;}

    public static class BuildException extends ConsentLibException {
        public static final String description = "Error during ConsentLib build";

        BuildException(Throwable e, String message) { super(e, description + ": " + message); }
        BuildException(String message) { super(description + ": " + message); }
    }

    public static class NoInternetConnectionException extends ConsentLibException {
        public static final String description = "Error due to no internet connection";

        NoInternetConnectionException(Throwable e, String message) { super(e, description + ": " + message); }
        NoInternetConnectionException(String message) { super(description + ": " + message); }

    }

    public static class ApiException extends ConsentLibException {
        public static final String description = "Error due to android API";
        ApiException(Throwable e, String message) { super(e, description + ": " + message); }
        ApiException(String message) { super(description + ": " + message); }
    }
}

File: kotlin_sdk/cmplibrary/src/main/java/com/sourcepointmeta/cmplibrary/ConsentLibException.kt
package com.sourcepointmeta.cmplibrary

open class ConsentLibException : Exception {

    constructor() : super() {}
    constructor(message: String?) : super(message) {}

    class BuildException internal constructor(message: String?) :
        ConsentLibException("Error during ConsentLib build: $message")

    class NoInternetConnectionException : ConsentLibException()
    class ApiException internal constructor(message: String) : ConsentLibException(message)
}